src/main/java/uk/gov/hmcts/divorce/caseworker/service/notification/FinalOrderGrantedNotification.java

```diff
@@ -39,6 +42,27 @@ public class FinalOrderGrantedNotification implements ApplicantNotification {
     @Autowired
     private CommonContent commonContent;
 
+    @Autowired
+    private FinalOrderGrantedPrinter printer;
+
+    @Override
+    public void sendToApplicant1Solicitor(final CaseData caseData, final Long caseId) {
+        log.info(FINAL_ORDER_GRANTED_NOTIFICATION_TO_FOR_CASE_ID,
+            caseData.getApplicationType().isSole() ? "applicant solicitor" : "applicant 1 solicitor", caseId);
+        notificationService.sendEmail(
+            caseData.getApplicant1().getSolicitor().getEmail(),
+            SOLICITOR_FINAL_ORDER_GRANTED,
+            solicitorTemplateContent(caseData, caseId, caseData.getApplicant1(), caseData.getApplicant2()),
+            caseData.getApplicant1().getLanguagePreference()
+        );
+    }
+
+    @Override
+    public void sendToApplicant1Offline(final CaseData caseData, final Long caseId) {
+        log.info("Sending Final Order Granted letter to Applicant 1: {}", caseId);
```

Comment:
Could we move the generation of docs from the aboutToSubmit into these methods? As there would be no need for any if checks and would clean it up (if I understand correctly 😂 )

Reply:
Unfortunately any documents generated in the submitted callback are not saved, so we need to do this before the submitted.